src/megatron/energon/flavors/webdataset/structs.py

```diff
@@ -13,7 +13,9 @@
 class WebdatasetInfo:
     """Info about a webdataset. Format for `.nv-meta/.info.yaml` or `.nv-meta/.info.json`."""
 
-    # Maps shard name to number of samples in that shard
+    #: The version of the energon library that was used to prepare the dataset
+    energon_version: Optional[str] = None
```

Comment:
Is `None` a good default? If we compare this in the future (i.e. if dataset.energon_version > ...), it'll require a special case. Maybe we should use `packaging.version.Version` directly to support comparisons properly.